packages/dds/tree/src/feature-libraries/editable-tree/editableTreeContext.ts

```diff
@@ -55,13 +69,39 @@ export interface EditableTreeContext {
      * EditableTrees created in this context are invalid to use after this.
      */
     free(): void;
+
+    /**
+     * Attaches the handler to be called after a transaction, initiated by
+     * either this context or any other context or client using this document,
+     * is committed successfully.
+     */
+    attachAfterChangeHandler(afterChangeHandler: (this: EditableTreeContext) => void): void;
 }
 
 export class ProxyContext implements EditableTreeContext {
     public readonly withCursors: Set<ProxyTarget<Anchor | FieldAnchor>> = new Set();
     public readonly withAnchors: Set<ProxyTarget<Anchor | FieldAnchor>> = new Set();
+    private readonly observer: Dependent;
+    private readonly afterChangeHandlers: Set<(this: EditableTreeContext) => void> = new Set();
 
-    constructor(public readonly forest: IEditableForest) {}
+    constructor(
```

Comment:
This should document why transactionCheckout is optional and what it means for it to be provided or not.

Reply:
Do you mean, even if it is documented for a function exposed as a public API?

Reply:
Done.